### ISSUE TITLE
Merge conflict when making consecutive calls to `create-pull-request` with `base` specified.

### ISSUE THREAD

**Issue by yardasol** [peter-evans/create-pull-request#1081]
### Subject of the issue
I am trying to use this action in a workflow that creates two PRs: One that will update the software version number when development begins and one that will change the `_version_extra` from `dev` to `0` to indicate the full release before we publish.

[Here](https://github.com/arfc/saltproc/runs/5399139681?check_suite_focus=true) is the workflow run and [here](https://github.com/arfc/saltproc/actions/runs/1924935644/workflow) is the workflow file

The first  use of the action executes without issue.
However, the second use of the action runs into a `git` error on the `Create or update the pull request branch` step:

```
  /usr/bin/git symbolic-ref HEAD --short
  v0.5.0-init
  Working base is branch 'v0.5.0-init'
  /usr/bin/git checkout --progress -B 7e02b4cf-619b-48f2-ac95-2256ef3c0596 HEAD --
  Switched to a new branch '7e02b4cf-619b-48f2-ac95-2256ef3c0596'
  M	doc/releasenotes/v0.5.0.rst
  M	saltproc/version.py
  /usr/bin/git status --porcelain -unormal
   M doc/releasenotes/v0.5.0.rst
   M saltproc/version.py
  Uncommitted changes found. Adding a commit.
  /usr/bin/git add -A
  /usr/bin/git -c author.name=yardasol -c author.email=yardasol@users.noreply.github.com -c committer.name=GitHub -c committer.email=noreply@github.com commit -m final changes before publishing v0.5.0
  [7e02b4cf-619b-48f2-ac95-2256ef3c0596 4373343] final changes before publishing v0.5.0
   Author: yardasol <yardasol@users.noreply.github.com>
   2 files changed, 3 insertions(+), 3 deletions(-)
  /usr/bin/git reset --hard
  HEAD is now at 4373343 final changes before publishing v0.5.0
  /usr/bin/git clean -f
  Resetting working base branch 'v0.5.0-init'
  /usr/bin/git checkout --progress v0.5.0-init --
  Switched to branch 'v0.5.0-init'
  /usr/bin/git reset --hard origin/v0.5.0-init
  HEAD is now at 0f0cc62 inital changes for v0.5.0
  Rebasing commits made to branch 'v0.5.0-init' on to base branch 'master'
  /usr/bin/git -c protocol.version=2 fetch --no-tags --progress --no-recurse-submodules --unshallow --force origin master:master
  remote: Enumerating objects: 6618, done.        
  remote: Counting objects:   0% (1/6504) 
  .
  .
  .
  Resolving deltas: 100% (4517/4517), completed with 72 local objects.
  /usr/bin/git checkout --progress master --
  Switched to branch 'master'
  Your branch is up to date with 'origin/master'.
  /usr/bin/git rev-list --reverse v0.5.0-init..7e02b4cf-619b-48f2-ac95-2256ef3c0596 .
  4373343b22b1d96eb536019b981f0c5216a5b743
  /usr/bin/git -c author.name=yardasol -c author.email=yardasol@users.noreply.github.com -c committer.name=GitHub -c committer.email=noreply@github.com cherry-pick --strategy=recursive --strategy-option=theirs 4373343b22b1d96eb536019b981f0c5216a5b743
  Auto-merging saltproc/version.py
  CONFLICT (modify/delete): doc/releasenotes/v0.5.0.rst deleted in HEAD and modified in 4373343 (final changes before publishing v0.5.0). Version 4373343 (final changes before publishing v0.5.0) of doc/releasenotes/v0.5.0.rst left in tree.
  error: could not apply 4373343... final changes before publishing v0.5.0
  hint: After resolving the conflicts, mark them with
  hint: "git add/rm <pathspec>", then run
  hint: "git cherry-pick --continue".
  hint: You can instead skip this commit with "git cherry-pick --skip".
  hint: To abort and get back to the state before "git cherry-pick",
  hint: run "git cherry-pick --abort".
  Error: Unexpected error: error: could not apply 4373343... final changes before publishing v0.5.0
  hint: After resolving the conflicts, mark them with
  hint: "git add/rm <pathspec>", then run
  hint: "git cherry-pick --continue".
  hint: You can instead skip this commit with "git cherry-pick --skip".
  hint: To abort and get back to the state before "git cherry-pick",
  hint: run "git cherry-pick --abort".
```
It looks like what is happening is that the changes from the branch that the first PR references are rebased to `master` for the second call, but then this causes a merge conflict. Not sure if this is a bug in the action or some specific issue with my repo. Here's some info I think may be relevant to fixing this:
- I make the version number changes to files in a workflow step (let's call it 1), then use the `create-pull-request` action in the next step (2) to create a PR with those changes. I then repeat 1 and 2 but for the pre-release changes to the version number.
- In the second call to `create-pull-request`, I specify the `base` as `master`. I did not do this in the first call to `create-pull-request`. The reason for this is that in my testing, when I did not specify the `base`, the PR pointed to the branch created in the first call to `create-pull-request`.
- The `branch` field for the first and second call to `create-pull-request` are different
- The `delete-branch` field is set to `true` for both calls to `create-pull-request`

### Steps to reproduce

See the [log](https://github.com/arfc/saltproc/runs/5399139681?check_suite_focus=true) and [workflow file](https://github.com/arfc/saltproc/actions/runs/1924935644/workflow)

**Comment by peter-evans** [comment]
Hi @yardasol 

After the action runs, the local git repository will be left on the created branch. So running it a second time without reinitialising the local repo would cause problems.

If you want to make to pull requests in the same job then I highly recommend running checkout again to reset the local repo.
```
      - uses: actions/checkout@v2

      # Make changes to pull request here

      - name: Create Pull Request
        uses: peter-evans/create-pull-request@v3

      - uses: actions/checkout@v2

      # Make changes to pull request here

      - name: Create Pull Request
        uses: peter-evans/create-pull-request@v3
```
However, a better approach would probably be to separate the creation of each pull request into its own job in the workflow.

**Comment by yardasol** [comment]
Good suggestion @peter-evans! I've moved the steps where I create the PRs to their own jobs. I still want the changes from the branch created by the first call to `create-pull-request` to be passed to the second PR (so that I don't have to repeat any of those instructions), so I've added a `git pull` step that pulls those changes local `master` branch on the runner. We then apply the further changes I want in the 2nd PR and run `create-pull-request`, and it works like a charm!
